### ISSUE TITLE
Restarting HA takes very long with MA enabled

### ISSUE THREAD

**Issue by gieljnssns** [music-assistant/hass-music-assistant#997]
### What version of Music Assistant has the issue?

2022.11.0

### The problem

When I enable MA in HA restarting HA takes more then 4 minutes.
This step takes very long
<img width="1139" alt="Schermafbeelding 2022-11-17 om 15 39 02" src="https://user-images.githubusercontent.com/17234951/202664926-0ce18c0a-863f-4b74-a643-6a32701fc931.png">

When I disable MA 
<img width="432" alt="Schermafbeelding 2022-11-18 om 10 11 42" src="https://user-images.githubusercontent.com/17234951/202665481-70696425-dece-45b3-b7b6-633fc237cea2.png">
Then the first time restarting HA still takes long, but from then HA restarts in less then a minute

### How to reproduce

Enable/disable MA

### Relevant log output

None

### Additional information

<img width="460" alt="Schermafbeelding 2022-11-18 om 10 15 56" src="https://user-images.githubusercontent.com/17234951/202666054-6dde9214-2a88-44ac-be01-3522b444a1a5.png">
This are all sonos players

<img width="526" alt="Schermafbeelding 2022-11-18 om 10 16 18" src="https://user-images.githubusercontent.com/17234951/202666108-f3754e21-0b4b-4bba-a762-202ee1a6ddcb.png">

<img width="533" alt="Schermafbeelding 2022-11-18 om 10 16 30" src="https://user-images.githubusercontent.com/17234951/202666131-589bf969-ad29-4db5-a80f-b61b5f1080c3.png">

### What version of Home Assistant Core are your running

2022.11.3

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by gieljnssns** [comment]
This isn't the cause. I'm running v2022.11 from when it is available.
And i have restarted HA a lot since then.

**Comment by erkr** [comment]
Both the Spotify and TuneIn configure correctly? So both are working after startup.

**Comment by gieljnssns** [comment]
> Both Spotify and TuneIn configured correctly? So both are working once MA started?!

Yes they are both working

**Comment by erkr** [comment]
I don't have a clue. Maybe someone else? I have a similar configuration with tunein, Spotify and a huge local collection. The startup time for my setup is just 2-3 seconds.

**Comment by OzGav** [comment]
The OP has previously had an issue with a player blocking the start. Perhaps try removing all but one of your players and see if that makes a difference. Try and pick a known working player?
Then I would suggest disabling the music providers one at a time and see if that makes a difference.

**Comment by gieljnssns** [comment]
You are referring this issue https://github.com/music-assistant/hass-music-assistant/issues/784
This was because i had the spotify media player from HA as a media player in MA.
Now I don't have Spotify enabled in HA anymore.
Since then I have also removed the MA integration (also the .db file) and re-installed MA.

**Comment by erkr** [comment]
Hi
Normally MA doesn't block startup. Most sync actions are performed after startup in background tasks.
I only expect this to happen with failing credentials for music providers or network issues to resources addressed by MA (I.e media players).
Without any logging I don't see ways to follow up on this. 
Maybe enabling debug logging for MA reveals some info:
``` 
logger:
  default: warning
  logs:
 Music Assistant:
    custom_components.mass: debug
    music_assistant: debug
``` 
Best Eric

**Comment by OzGav** [comment]
Hi @gieljnssns. Have you made any progress narrowing down what is causing the issue?

**Comment by gieljnssns** [comment]
I thought to wait until the new version is out…

**Comment by OzGav** [comment]
You certainly can although this issue might be a player that isn’t being worked on….

**Comment by gieljnssns** [comment]
[mass-debug.log](https://github.com/music-assistant/hass-music-assistant/files/10164401/mass-debug.log)

**Comment by erkr** [comment]
It looks like MA is roughly  started in 10 seconds, then some other part is starting
```
2022-12-06 09:17:31.391 WARNING (MainThread) [homeassistant.helpers.entity] Update of calendar.rafc is taking over 10 seconds
```

The next MA logging is from some automation I Guess, after 10 minutes delay:
``` 
2022-12-06 09:28:50.681 DEBUG (MainThread) [custom_components.mass.player_controls.media_player.keuken] volume_set command called with value: 65

``` 

I doubth MA is not starting gaat

**Comment by gieljnssns** [comment]
@erkr 
I think it is the shutting down of MA. 
this step takes very long

![](https://user-images.githubusercontent.com/17234951/202664926-0ce18c0a-863f-4b74-a643-6a32701fc931.png)

Then the music stops playing and HA restarts

**Comment by erkr** [comment]
I indeed also noticed that during a restart, MA continues to play until HA starts up again.
But that doesn't delay my startup at all. Also at the bottom of your log, or looks like MA stopped within a second:
``` 
2022-12-06 09:40:59.782 INFO (MainThread) [music_assistant] Stop called, cleaning up...
2022-12-06 09:40:59.783 DEBUG (MainThread) [music_assistant.event] application_shutdown 
2022-12-06 09:40:59.783 INFO (MainThread) [music_assistant.db] Database disconnected.
2022-12-06 09:40:59.815 DEBUG (MainThread) [music_assistant.stream] Streamserver exited.
``` 
What happens if you only reload the MA integration while playing music? Same delay of several  minutes? Or does it just take some seconds?

**Comment by gieljnssns** [comment]
Sorry, will try when the MA version for HA 2022.12 is out.

**Comment by marcelveldt** [comment]
Restarting or reloading HA while music is playing is what cause the delay due to buffers that need to be freed. A signal is sent but not consumed as fast as we may wish.

Another reason we should move the core of MA to a separate add-on.

**Comment by gieljnssns** [comment]
Reloading the integration takes a couple seconds.

**Comment by gieljnssns** [comment]
There must be something in MA that is blocking a restart  👆🏻

**Comment by erkr** [comment]
The MA 2.0 release will move all back end stuff to an add-on, in order to be more independent of HA. Let's wait for that launch and see if your issue persists.

**Comment by marcelveldt** [comment]
This issue is fixed in v2.0 release because the entire corer of MA then runs in its own dedicated process.

**Comment by OzGav** [comment]
@gieljnssns I see you have moved to the beta so I am closing this.
